### ISSUE TITLE
[posgresql] Upgrade from 4.3.8 to 4.5.0 fails

### ISSUE THREAD

**Issue by skibbipl** [firefly-iii/firefly-iii#667]
I still have issues migrating to newest firefly version (see #644). `php artisan firefly:upgrade-database` gives me following error:
```
[2017-06-08 08:45:50] production.ERROR: PDOException: SQLSTATE[42883]: Undefined function: 7 ERROR:  operator does not exist: integer <> text
LINE 1: ... "transaction_journals"."transaction_currency_id" != account...
                                                             ^
HINT:  No operator matches the given name and argument type(s). You might need to add explicit type casts. in /var/www/firefly-dev/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php:91
Stack trace:
#0 /var/www/firefly-dev/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php(91): PDOStatement->execute(NULL)
#1 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Database/Connection.php(323): Doctrine\DBAL\Driver\PDOStatement->execute()
#2 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Database/Connection.php(640): Illuminate\Database\Connection->Illuminate\Database\{closure}('select "transac...', Array)
#3 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Database/Connection.php(607): Illuminate\Database\Connection->runQueryCallback('select "transac...', Array, Object(Closure))
#4 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Database/Connection.php(326): Illuminate\Database\Connection->run('select "transac...', Array, Object(Closure))
#5 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(1710): Illuminate\Database\Connection->select('select "transac...', Array, true)
#6 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(1695): Illuminate\Database\Query\Builder->runSelect()
#7 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(462): Illuminate\Database\Query\Builder->get(Array)
#8 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(446): Illuminate\Database\Eloquent\Builder->getModels(Array)
#9 /var/www/firefly-dev/app/Console/Commands/UpgradeDatabase.php(338): Illuminate\Database\Eloquent\Builder->get(Array)
#10 /var/www/firefly-dev/app/Console/Commands/UpgradeDatabase.php(75): FireflyIII\Console\Commands\UpgradeDatabase->updateJournalCurrencies()
#11 [internal function]: FireflyIII\Console\Commands\UpgradeDatabase->handle()
#12 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(29): call_user_func_array(Array, Array)
#13 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(87): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#14 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(31): Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application), Array, Object(Closure))
#15 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Container/Container.php(539): Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application), Array, Array, NULL)
#16 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Console/Command.php(182): Illuminate\Container\Container->call(Array)
#17 /var/www/firefly-dev/vendor/symfony/console/Command/Command.php(264): Illuminate\Console\Command->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#18 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Console/Command.php(167): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#19 /var/www/firefly-dev/vendor/symfony/console/Application.php(869): Illuminate\Console\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#20 /var/www/firefly-dev/vendor/symfony/console/Application.php(223): Symfony\Component\Console\Application->doRunCommand(Object(FireflyIII\Console\Commands\UpgradeDatabase), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#21 /var/www/firefly-dev/vendor/symfony/console/Application.php(130): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#22 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(122): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#23 /var/www/firefly-dev/artisan(35): Illuminate\Foundation\Console\Kernel->handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#24 {main}

Next Doctrine\DBAL\Driver\PDOException: SQLSTATE[42883]: Undefined function: 7 ERROR:  operator does not exist: integer <> text
LINE 1: ... "transaction_journals"."transaction_currency_id" != account...
                                                             ^
HINT:  No operator matches the given name and argument type(s). You might need to add explicit type casts. in /var/www/firefly-dev/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php:93
Stack trace:
#0 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Database/Connection.php(323): Doctrine\DBAL\Driver\PDOStatement->execute()
#1 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Database/Connection.php(640): Illuminate\Database\Connection->Illuminate\Database\{closure}('select "transac...', Array)
#2 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Database/Connection.php(607): Illuminate\Database\Connection->runQueryCallback('select "transac...', Array, Object(Closure))
#3 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Database/Connection.php(326): Illuminate\Database\Connection->run('select "transac...', Array, Object(Closure))
#4 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(1710): Illuminate\Database\Connection->select('select "transac...', Array, true)
#5 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(1695): Illuminate\Database\Query\Builder->runSelect()
#6 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(462): Illuminate\Database\Query\Builder->get(Array)
#7 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(446): Illuminate\Database\Eloquent\Builder->getModels(Array)
#8 /var/www/firefly-dev/app/Console/Commands/UpgradeDatabase.php(338): Illuminate\Database\Eloquent\Builder->get(Array)
#9 /var/www/firefly-dev/app/Console/Commands/UpgradeDatabase.php(75): FireflyIII\Console\Commands\UpgradeDatabase->updateJournalCurrencies()
#10 [internal function]: FireflyIII\Console\Commands\UpgradeDatabase->handle()
#11 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(29): call_user_func_array(Array, Array)
#12 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(87): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#13 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(31): Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application), Array, Object(Closure))
#14 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Container/Container.php(539): Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application), Array, Array, NULL)
#15 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Console/Command.php(182): Illuminate\Container\Container->call(Array)
#16 /var/www/firefly-dev/vendor/symfony/console/Command/Command.php(264): Illuminate\Console\Command->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#17 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Console/Command.php(167): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#18 /var/www/firefly-dev/vendor/symfony/console/Application.php(869): Illuminate\Console\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#19 /var/www/firefly-dev/vendor/symfony/console/Application.php(223): Symfony\Component\Console\Application->doRunCommand(Object(FireflyIII\Console\Commands\UpgradeDatabase), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#20 /var/www/firefly-dev/vendor/symfony/console/Application.php(130): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#21 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(122): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#22 /var/www/firefly-dev/artisan(35): Illuminate\Foundation\Console\Kernel->handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#23 {main}

Next Illuminate\Database\QueryException: SQLSTATE[42883]: Undefined function: 7 ERROR:  operator does not exist: integer <> text
LINE 1: ... "transaction_journals"."transaction_currency_id" != account...
                                                             ^
HINT:  No operator matches the given name and argument type(s). You might need to add explicit type casts. (SQL: select "transaction_journals".*, "account_meta"."data" as "expected_currency_id", "transactions"."amount" as "transaction_amount" from "transaction_journals" left join "transaction_types" on "transaction_types"."id" = "transaction_journals"."transaction_type_id" left join "transactions" on "transaction_journals"."id" = "transactions"."transaction_journal_id" and "transactions"."amount" < 0 left join "accounts" on "accounts"."id" = "transactions"."account_id" left join "account_meta" on "account_meta"."account_id" = "accounts"."id" where "transaction_types"."type" = Withdrawal and "account_meta"."name" = currency_id and "transaction_journals"."transaction_currency_id" != account_meta.data and "transaction_journals"."deleted_at" is null) in /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Database/Connection.php:647
Stack trace:
#0 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Database/Connection.php(607): Illuminate\Database\Connection->runQueryCallback('select "transac...', Array, Object(Closure))
#1 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Database/Connection.php(326): Illuminate\Database\Connection->run('select "transac...', Array, Object(Closure))
#2 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(1710): Illuminate\Database\Connection->select('select "transac...', Array, true)
#3 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(1695): Illuminate\Database\Query\Builder->runSelect()
#4 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(462): Illuminate\Database\Query\Builder->get(Array)
#5 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(446): Illuminate\Database\Eloquent\Builder->getModels(Array)
#6 /var/www/firefly-dev/app/Console/Commands/UpgradeDatabase.php(338): Illuminate\Database\Eloquent\Builder->get(Array)
#7 /var/www/firefly-dev/app/Console/Commands/UpgradeDatabase.php(75): FireflyIII\Console\Commands\UpgradeDatabase->updateJournalCurrencies()
#8 [internal function]: FireflyIII\Console\Commands\UpgradeDatabase->handle()
#9 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(29): call_user_func_array(Array, Array)
#10 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(87): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#11 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(31): Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application), Array, Object(Closure))
#12 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Container/Container.php(539): Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application), Array, Array, NULL)
#13 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Console/Command.php(182): Illuminate\Container\Container->call(Array)
#14 /var/www/firefly-dev/vendor/symfony/console/Command/Command.php(264): Illuminate\Console\Command->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#15 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Console/Command.php(167): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#16 /var/www/firefly-dev/vendor/symfony/console/Application.php(869): Illuminate\Console\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#17 /var/www/firefly-dev/vendor/symfony/console/Application.php(223): Symfony\Component\Console\Application->doRunCommand(Object(FireflyIII\Console\Commands\UpgradeDatabase), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#18 /var/www/firefly-dev/vendor/symfony/console/Application.php(130): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#19 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(122): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#20 /var/www/firefly-dev/artisan(35): Illuminate\Foundation\Console\Kernel->handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#21 {main}



  [Illuminate\Database\QueryException]
  SQLSTATE[42883]: Undefined function: 7 ERROR:  operator does not exist: integer <> text
  LINE 1: ... "transaction_journals"."transaction_currency_id" != account...
                                                               ^
  HINT:  No operator matches the given name and argument type(s). You might need to add explicit type casts. (SQL: select "transaction_journals".*, "account_meta"."data" as "expected_currency_id", "transa
  ctions"."amount" as "transaction_amount" from "transaction_journals" left join "transaction_types" on "transaction_types"."id" = "transaction_journals"."transaction_type_id" left join "transactions" on
  "transaction_journals"."id" = "transactions"."transaction_journal_id" and "transactions"."amount" < 0 left join "accounts" on "accounts"."id" = "transactions"."account_id" left join "account_meta" on "a
  ccount_meta"."account_id" = "accounts"."id" where "transaction_types"."type" = Withdrawal and "account_meta"."name" = currency_id and "transaction_journals"."transaction_currency_id" != account_meta.dat
  a and "transaction_journals"."deleted_at" is null)



  [Doctrine\DBAL\Driver\PDOException]
  SQLSTATE[42883]: Undefined function: 7 ERROR:  operator does not exist: integer <> text
  LINE 1: ... "transaction_journals"."transaction_currency_id" != account...
                                                               ^
  HINT:  No operator matches the given name and argument type(s). You might need to add explicit type casts.



  [PDOException]
  SQLSTATE[42883]: Undefined function: 7 ERROR:  operator does not exist: integer <> text
  LINE 1: ... "transaction_journals"."transaction_currency_id" != account...
                                                               ^
  HINT:  No operator matches the given name and argument type(s). You might need to add explicit type casts.
```

**Comment by JC5** [comment]
Damn. Go to line 319 in app/Console/Commands/UpgradeDatabase.php and enter this:
```
var_dump($driver);
exit;
```

Could you tell me what it says?

**Comment by skibbipl** [comment]
`string(5) "pgsql"`

**Comment by JC5** [comment]
Alright. I've updated the file in the development branch, and I would ask you to copy the changes so your upgrade will continue as planned:

https://github.com/firefly-iii/firefly-iii/commit/a2145f6b490e77422d02aed0600fdb6c1732d212

**Comment by skibbipl** [comment]
Nope:
```
[2017-06-08 08:57:42] production.ERROR: Exception is: {"class":"ErrorException","errorMessage":"Undefined variable: pgsql","time":"Thu, 08 Jun 2017 08:57:42 +0000","file":"\/var\/www\/firefly-dev\/app\/Console\/Commands\/UpgradeDatabase.php","line":331,"code":0,"version":"4.5.0"}

[2017-06-08 08:57:42] production.ERROR: ErrorException: Undefined variable: pgsql in /var/www/firefly-dev/app/Console/Commands/UpgradeDatabase.php:331
Stack trace:
#0 /var/www/firefly-dev/app/Console/Commands/UpgradeDatabase.php(331): Illuminate\Foundation\Bootstrap\HandleExceptions->handleError(8, 'Undefined varia...', '/var/www/firefl...', 331, Array)
#1 /var/www/firefly-dev/app/Console/Commands/UpgradeDatabase.php(75): FireflyIII\Console\Commands\UpgradeDatabase->updateJournalCurrencies()
#2 [internal function]: FireflyIII\Console\Commands\UpgradeDatabase->handle()
#3 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(29): call_user_func_array(Array, Array)
#4 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(87): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#5 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(31): Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application), Array, Object(Closure))
#6 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Container/Container.php(539): Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application), Array, Array, NULL)
#7 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Console/Command.php(182): Illuminate\Container\Container->call(Array)
#8 /var/www/firefly-dev/vendor/symfony/console/Command/Command.php(264): Illuminate\Console\Command->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#9 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Console/Command.php(167): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#10 /var/www/firefly-dev/vendor/symfony/console/Application.php(869): Illuminate\Console\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#11 /var/www/firefly-dev/vendor/symfony/console/Application.php(223): Symfony\Component\Console\Application->doRunCommand(Object(FireflyIII\Console\Commands\UpgradeDatabase), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#12 /var/www/firefly-dev/vendor/symfony/console/Application.php(130): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#13 /var/www/firefly-dev/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(122): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#14 /var/www/firefly-dev/artisan(35): Illuminate\Foundation\Console\Kernel->handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#15 {main}



  [ErrorException]
  Undefined variable: pgsql
```

**Comment by JC5** [comment]
Did you add this line as well?

```
$pgsql        = ['pgsql', 'postgresql'];
```

**Comment by skibbipl** [comment]
Yes I forgot, but it didn't help. I have the same error as before. It seems the postgresql migration script is bad. The errors are related to database operations.

**Comment by JC5** [comment]
Damn. I know the script is bad, I was hoping that the cast would work. I'll set up  a postgresql machine myself, to debug this properly. I'll keep you posted!

**Comment by JC5** [comment]
As far as I can tell, the cast works. My own server upgraded without a hitch and it hit the cast a few times. I'll keep you posted.

**Comment by skibbipl** [comment]
If it can help I have few accounts with different currency declared. Maybe when you have multiple accounts with different currencies the upgrade procedure fails?

**Comment by JC5** [comment]
It shouldn't, but it's certainly possible. I'll extend my test case and post some screenshots. If you could let me know if it compares to your situation.

This is the database:

![screen shot 2017-06-09 at 08 13 19](https://user-images.githubusercontent.com/5889984/26963187-a7070c74-4ceb-11e7-9968-bf90d9d40478.png)

And this is the result from the upgrade script:

```
vagrant@vagrant-ubuntu-trusty-64:/sites/FF3/firefly-iii$ php artisan firefly:upgrade-database
Account #1 ("Checking Account") now has a currency setting (EUR).
Account #3 ("Shared Checking Account") now has a currency setting (EUR).
Account #32 ("Checking Account X") now has a currency setting (EUR).
Account #37 ("2 Hungarian forint account") now has a correct currency for opening balance.
Account #38 ("3 Real account") now has a currency setting (EUR).
Account #39 ("4 Will be EUR account") now has a currency setting (EUR).
```
At this point _more_ accounts will have a currency setting:

![screen shot 2017-06-09 at 08 16 15](https://user-images.githubusercontent.com/5889984/26963254-ee4b8218-4ceb-11e7-9818-fca7b059307b.png)

But the script continues:

```
Withdrawal #600 uses HUF but should use EUR. It has been updated. Please verify this in Firefly III.
Withdrawal #601 uses BRL but should use EUR. It has been updated. Please verify this in Firefly III.
Deposit #603 uses GBP but should use EUR. It has been updated. Please verify this in Firefly III.
Transfer #605 has been updated to use the correct currencies. Please verify this in Firefly III.
Transfer #606 has been updated to use the correct currencies. Please verify this in Firefly III.
Transfer #607 has been updated to use the correct currencies. Please verify this in Firefly III.
Updated currency information for 1232 transactions
Updated currency information for 0 transactions
Firefly III database is up to date.
vagrant@vagrant-ubuntu-trusty-64:/sites/FF3/firefly-iii$
```

It seems to work. Is this comparable to your installation?

**Comment by skibbipl** [comment]
Did you change anything in the code? I still cannot upgrade.

**Comment by JC5** [comment]
I'm not sure what the status is. What error are you getting at the moment?

**Comment by skibbipl** [comment]
I just saw, that You released 4.5.1. I'll try to upgrade tomorrow to see if it helps.
